app/types/datetime.go

```diff
@@ -0,0 +1,76 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+)
+
+// Doc is mainly in the "int64" file :-)
+
+type (
+	// Datetime is a date+time format which can be set/not-set and null/not-null
+	Datetime struct {
+		Value time.Time
+		Set   bool
+		Null  bool
+	}
+	// RequiredDatetime must be set and not null
+	RequiredDatetime struct{ Datetime }
+	// NullableDatetime must be set and can be null
+	NullableDatetime struct{ Datetime }
+	// OptionalDatetime can be not set. If set, cannot be null.
+	OptionalDatetime struct{ Datetime }
+	// OptNullDatetime can be not set or null
+	OptNullDatetime struct{ Datetime }
+)
+
+// NewDatetime creates a Datetime which is not-null and set with the given value
+func NewDatetime(t time.Time) *Datetime {
```

Comment:
Ditto, SQL clients are capable of operating on `time.Time` values, there should be no need of creating such types.

Reply:
https://github.com/go-sql-driver/mysql/blob/749ddf1598b47e3cd909414bda735fe790ef3d30/utils.go#L123

Reply:
this is not related with that, this is for the same reason we have the types/int64, to be able to distinguish between not set, set to default and set to null, which all arrive as the same value in Go.

Reply:
You have a built-in type for that as well: https://github.com/golang/go/blob/master/src/database/sql/sql.go#L214

Reply:
Still mixing a null value given and "not set".

Reply:
Actually, this is an interesting debate but a bit out of scope of this PR this was introduced earlier. I see this wrapped type is not convenient in Go, but I still find ridiculous to drop a sound feature because of a language stupidity (personnal opinion 😁).